### PR TITLE
Gorelease Added

### PR DESCRIPTION
**Pull request by evalsocket**
## Read then delete

# TL;DR
1. Goreleaser added for managing kubectl-flyte build and release.  
2. Github action added for releasing the binary 
3. Added version package 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/120

## Follow-up issue
https://github.com/lyft/flyte/issues/120

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+project_name: kubectl-flyte
+before:
+  hooks:
+    - go mod download
+builds:
+  - env:
+      - CGO_ENABLED=0
+    main: ./cmd/kubectl-flyte/main.go
+    binary: kubectl-flyte
```
Comment (evalsocket) [inline .goreleaser.yml:9]: @kumare3 Can you suggest a name for plugin name. https://krew.sigs.k8s.io/docs/developer-guide/develop/naming-guide/

File: .krew.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: kubectl-flyte
+spec:
+  version: "{{ .TagName }}"
+  homepage: https://github.com/lyft/flytepropeller
+  shortDescription: >-
```
Comment (evalsocket) [inline .krew.yaml:8]: `kubectl kubectl-flyte` is not a right way to communicate can we use some other name. Because when user is using it with kubectl then intend is clear, we don't need kubectl to specify the intend

**Comment by kumare3** [comment]
cc @EngHabu who added goreleaser to flytestdlib

File: .goreleaser.yml (hunk 1 above)
Comment (EngHabu) [reply .goreleaser.yml:9]: I think this is fine... we do want the usage to be `kubectl flyte get -n my-namespace` to retrieve flyte workflows

File: .krew.yaml (hunk 2 above)
Comment (EngHabu) [reply .krew.yaml:8]: my understanding is kubectl uses this convention to "discover" plugins, the binary should be called `kubectl-SOMETHING` and it then allows you to call `kubectl SOMETHING get... ` so the usage description here should say `kubectl flyte get -n my-namespace` to retrieve workflows.... etc.

File: .goreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    name: goreleaserbot
+    email: goreleaser@carlosbecker.com
+
+  # Your app's homepage.
+  # Default is empty.
+  homepage: "https://godoc.org/github.com/lyft/flytepropeller"
+
+  # Your app's description.
+  # Default is empty.
+  description: "Common Go utilities (Typed-Config, PFlags, Prometheus Metrics,... more)."
```
Comment (EngHabu) [inline .goreleaser.yml:55]: Can you update this to reflect what it does...

File: .krew.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  description: |+2
+      kubectl-flyte is an command line tool that can be used as an extension to kubectl.
```
Comment (EngHabu) [inline .krew.yaml:10]:
```suggestion
      kubectl-flyte is an command line tool that can be used as an extension to kubectl. It offers an easier way to interact with flyte custom resource definitions on the cluster.
```

or something?

**Comment by evalsocket** [comment]
@kumare3 @EngHabu I removed krew changes from this PR. There is one constrain, for krew plugin publish we need to do it manually the first time and for creating PR in krew we need our binary in GitHub release

File: version/version.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// This module provides the ability to inject Build (git sha) and Version information at compile time.
+// To set these values invoke go build as follows
+// go build -ldflags “-X github.com/lyft/flytepropeller/version.Build=xyz -X github.com/lyft/flytepropeller/version.Version=1.2.3"
+// NOTE: If the version is set and server.StartProfilingServerWithDefaultHandlers are initialized then, `/version`
```
Comment (EngHabu) [inline version/version.go:12]: I don't think this comment is correct... it's only correct if you use the stdlib's version of these vars :)

Comment (evalsocket) [reply version/version.go:12]: I used the same package. And I missed it

**Comment by EngHabu** [comment]
I'm trying to figure out the best setup to make github actions run. I understand they won't run here but I think if you merge your changes to your own master, they should (e.g. https://github.com/evalsocket/flytepropeller/compare/evalsocket:master...evalsocket:fix-121?expand=1)

You will need to set a github secret in your fork called `flytegithub_repo` with value `evalsocket/flytepropeller` to be able to build and push images to your github docker registry.

**Comment by evalsocket** [comment]
@EngHabu I think, I know what's wrong there 
```
on:
  push:
    branches:
      - 'master'
    tags:
      - 'v*'
  pull_request:
```

**Comment by evalsocket** [comment]
@EngHabu please review it https://github.com/lyft/flytepropeller/pull/170
